app/code/Magento/Reports/Test/Unit/Observer/CheckoutCartAddProductObserverTest.php

```diff
@@ -0,0 +1,166 @@
+            ->disableOriginalConstructor()
+            ->setMethods(['getItem'])
+            ->getMock();
+
+        $objectManager = new ObjectManager($this);
+        $this->observer = $objectManager->getObject(
+            CheckoutCartAddProductObserver::class,
+            [
+                'eventSaver' => $this->eventSaverMock,
+                'reportStatus' => $this->reportStatusMock
+            ]
+        );
+    }
+
+    /**
+     * The case when event has to be successfully saved
+     */
+    public function testExecute()
+    {
+        $this->configureMocksWhenReportsEnabled();
+        $this->quoteItemMock->expects($this->once())
+            ->method('getId')
+            ->willReturn(null);
+        $this->quoteItemMock->expects($this->once())
+            ->method('getParentItem')
+            ->willReturn(null);
+
+        $this->eventSaverMock->expects($this->once())
+            ->method('save');
+        $this->observer->execute($this->eventObserverMock);
+    }
+
+    /**
+     * Test the method when 'Product Added To Cart' Report is disabled in configuration
+     */
+    public function testExecuteWhenReportsDisabled()
+    {
+        $this->reportStatusMock->expects($this->once())
+            ->method('isReportEnabled')
+            ->with(ReportsEventModel::EVENT_PRODUCT_TO_CART)
+            ->willReturn(false);
+
+        $this->checkOriginalMethodIsNeverExecuted();
+    }
+
+    /**
+     * Test when Quote Item has Id
+     */
+    public function testExecuteWithQuoteItemIdSet()
+    {
+        $this->configureMocksWhenReportsEnabled();
+        $this->quoteItemMock->expects($this->any())
+            ->method('getId')
+            ->willReturn(self::STUB_QUOTE_ITEM_ID);
+
+        $this->checkOriginalMethodIsNeverExecuted();
+    }
+
+    /**
+     * Test when Quote Item has Parent Item set
+     */
+    public function testExecuteWithQuoteParentItemIdSet()
+    {
+        $this->configureMocksWhenReportsEnabled();
+        $this->quoteItemMock->expects($this->any())
+            ->method('getParentItem')
+            ->willReturn(self::STUB_QUOTE_PARENT_ITEM_ID);
+
+        $this->checkOriginalMethodIsNeverExecuted();
+    }
+
+    /**
+     * Common mocks assertions when Report is enabled in configuration
+     */
+    private function configureMocksWhenReportsEnabled()
+    {
+        $this->reportStatusMock->expects($this->once())
+            ->method('isReportEnabled')
+            ->willReturn(true);
+        $this->eventObserverMock->expects($this->once())
+            ->method('getEvent')
+            ->willReturn($this->eventMock);
+        $this->eventMock->expects($this->once())
+            ->method('getItem')
+            ->willReturn($this->quoteItemMock);
+    }
+
+    /**
+     * Checking that the method will be never executed
+     */
+    private function checkOriginalMethodIsNeverExecuted()
+    {
+        $this->eventSaverMock->expects($this->never())
```

Comment:
That actually does not exceeds 80 characters in a row, so you can keep that single-line.